src/app/core/services/view.service.ts

```diff
@@ -114,18 +114,41 @@ export class ViewService {
   }
 
   /**
-   * Change repository if a valid repository is provided
-   * @param repo New repository
+   * Verifies if the organisation and repository are present on GitHub.
+   * @param org Organisation name
+   * @param repo Repository name
+   * @returns Promise that resolves to true if the organisation and repository are present.
+   * @throws Error if the organisation or repository are not present.
    */
-  async changeRepositoryIfValid(repo: Repo) {
-    this.isChangingRepo.next(true);
+  async verifyOwnerAndRepo(owner: string, repo: string): Promise<boolean> {
+    const isValidOwner =
+      (await this.githubService.isOrganisationPresent(owner).toPromise()) ||
+      (await this.githubService.isUsernamePresent(owner).toPromise());
+    if (!isValidOwner) {
+      throw new Error(ErrorMessageService.repoOwnerNotPresentMessage());
+    }
 
-    const isValidRepository = await this.githubService.isRepositoryPresent(repo.owner, repo.name).toPromise();
+    const isValidRepository = await this.githubService.isRepositoryPresent(owner, repo).toPromise();
     if (!isValidRepository) {
-      this.isChangingRepo.next(false);
       throw new Error(ErrorMessageService.repositoryNotPresentMessage());
     }
 
+    return true;
+  }
+
+  /**
+   * Change repository if a valid repository is provided
+   * @param repo New repository
+   * @throws Error if the repository is not valid
+   */
+  async changeRepositoryIfValid(repo: Repo) {
+    try {
+      this.isChangingRepo.next(true);
+      await this.verifyOwnerAndRepo(repo.owner, repo.name);
+    } catch (error) {
+      this.isChangingRepo.next(false);
+      throw error;
+    }
```

Comment:
Would not recommend try-catching as part of the control flow, i.e. `this.isChangingRepo.next(false)` should not be in the `catch` block. Perhaps you can do this in `verifyOwnerAndRepo` before throwing the error in the same method, and remove the try-catch block here?

Reply:
`verifyOwnerAndRepo` is used in another portion of the code though, which does not require the `this.isChangingRepo.next(false)` statement. 

Perhaps I can make `verifyOwnerAndRepo` return a constant string or enum? then i can throw the relevant errors from here instead. can do the same for the other usages of `verifyOwnerAndRepo` as well

Reply:
I believe it is fine to have that line of code in the method `initializeCurrentRepository`? Do try and let me know if it causes any unintended effect.

Reply:
hmm yeah should be okay, have commited and pushed the change